packages/metascraper-audio/index.js

```diff
@@ -43,28 +46,36 @@ const _getIframe = async (url, { src }) => {
   return iframe.document.documentElement.outerHTML
 }
 
-module.exports = ({ getIframe = _getIframe } = {}) => ({
+module.exports = ({ getIframe = _getIframe, gotOpts } = {}) => ({
   audio: [
     ...audioRules,
     async ({ htmlDom: $, url }) => {
-      const src = $filter($, $('iframe'), el =>
-        normalizeUrl(url, el.attr('src'))
-      )
+      const iframe = $('iframe')
+      if (iframe.length === 0) return
 
+      const src = $filter($, iframe, el => normalizeUrl(url, el.attr('src')))
       if (!src) return
 
       const html = await getIframe(url, { src })
       const htmlDom = cheerio.load(html)
 
-      let index = 0
-      let value
+      return findRule(audioRules, { htmlDom, url })
+    },
+    async ({ htmlDom: $, url }) => {
+      const playerUrl =
+        $('meta[name="twitter:player"]').attr('content') ||
+        $('meta[property="twitter:player"]').attr('content')
+
+      if (!playerUrl) return
 
-      do {
-        const rule = audioRules[index++]
-        value = await rule({ htmlDom, url })
-      } while (!has(value) && index < audioRules.length)
+      const { headers } = await got.head(playerUrl, gotOpts)
+      const contentType = headers['content-type']
+      if (!contentType || !contentType.startsWith('text')) return
+
+      const html = await got(playerUrl, { resolveBodyOnly: true, ...gotOpts })
```

Comment:
I haven't come across any instances where this would be helpful, but you could use `browserless` (i.e. the injected `getIframe` fn) to make it more likely this this url will load to completion. As is, if the audio url isn't present in the initial http response body then nothing will be found.

I don't think it's necessary to do this now, I'm just pointing it out as a possibility.

I'll let you know if I ever come across an example that could benefit from this change 😁

Reply:
That's a great point!

This is done intentionally in a simple way (just HTTP fetch) because this use case is very specific to a Twitter player stream.

Most sites are using `twitter:player` with a media source URL rather than hosting a player that has the media URL inside.

Also, get the content via browserless has some overload and I'm not sure if it's worth it.